### DISCUSSION TITLE
TimeZone utc to local and local to utc is missing.

### DISCUSSION THREAD

**Discussion by gopaldahal** [pnp/pnpcore#480] [Q&A] [answered]
Was exploring the date conversion from UTC to local and local to UTC. but with no luck.
https://s-kainet.github.io/sp-rest-explorer/#/entity/SP.TimeZone

conversion that is available presently on CSOM
-UTC to Local Time
-Local To UTC time

Equivalent Code on CSOM
var conversionType = DateTimeKind.Utc or DateTimeKind.Local;
var specifiedDate= DateTime.SpecifyKind(dtVal, conversionType );
Microsoft.SharePoint.Client.Utilities.Utility.FormatDateTime(context, context.Web, specifiedDate, Microsoft.SharePoint.Client.Utilities.DateTimeFormat.ISO8601);

**Comment by DaleyKD** [comment, marked as answer]
Can you be a bit more descriptive?

**Reply by gopaldahal** [reply]
I was searching for a similar conversion that is available presently on CSOM
-UTC to Local Time
-Local To UTC time

Code on CSOM
var conversionType = DateTimeKind.Utc or DateTimeKind.Local;
var specifiedDate= DateTime.SpecifyKind(dtVal, conversionType );
Microsoft.SharePoint.Client.Utilities.Utility.FormatDateTime(context, context.Web, specifiedDate, Microsoft.SharePoint.Client.Utilities.DateTimeFormat.ISO8601);

**Reply by jansenbe** [reply]
Tracked via https://github.com/pnp/pnpcore/issues/484

**Reply by jansenbe** [reply]
Implemented. Sample use:

```csharp
// convert to site's timezone
var localSiteTime = context.Web.RegionalSettings.TimeZone.UtcToLocalTime(utcDate);

// convert back to UTC time
var localSiteTimeBackToUtc = context.Web.RegionalSettings.TimeZone.LocalTimeToUtc(localSiteTime);
```
